pyaedt/edb_core/edb_data/hfss_simulation_setup_data.py

```diff
@@ -0,0 +1,1700 @@
+
+        Returns
+        -------
+        bool
+        """
+        return self.adaptive_settings.UseMaxRefinement
+
+    @use_max_refinement.setter
+    def use_max_refinement(self, value):
+        self.adaptive_settings.UseMaxRefinement = value
+        self._parent._update_setup()
+
+    @pyaedt_function_handler()
+    def add_adaptive_frequency_data(self, frequency, max_num_passes=10, max_delta_s=0.02):
+        """Add a Frequency Data setup.
+
+        Parameters
+        ----------
+        frequency : str, float
+            Frequency with units or float frequency (in Hz).
+        max_num_passes : int, optional
+            Maximum number of passes. Default is ``10``.
+        max_delta_s : float, optional
+            Maximum Delta S. Default is ``0.02``.
+
+        Returns
+        -------
+        bool
+        """
+        adaptive_frequency_data = self._parent._edb.simsetupdata.AdaptiveFrequencyData()
+        data = AdaptiveFrequencyData(adaptive_frequency_data)
+        data.adaptive_frequency = self._parent._edb.arg_with_dim(frequency, "Hz")
+        data.max_passes = max_num_passes
+        data.max_delta = str(max_delta_s)
+        self.adaptive_settings.AdaptiveFrequencyDataList.Add(data._adaptive_frequency_data)
+        return self._parent._update_setup()
+
+
+class DefeatureSettings(object):
+    """Manages EDB methods for defeature settings."""
+
+    def __init__(self, parent):
+        self._parent = parent
+
+    @property
+    def _defeature_settings(self):
+        return self._parent._edb_sim_setup_info.SimulationSettings.DefeatureSettings
+
+    @property
+    def defeature_abs_length(self):
+        """Defeature absolute length.
+
+        Returns
+        -------
+        str
+        """
+        return self._defeature_settings.DefeatureAbsLength
+
+    @defeature_abs_length.setter
+    def defeature_abs_length(self, value):
+        self._defeature_settings.DefeatureAbsLength = value
+        self._parent._update_setup()
+
+    @property
+    def defeature_ratio(self):
+        """Defeature ratio.
+
+        Returns
+        -------
+        float
+        """
+        return self._defeature_settings.DefeatureRatio
+
+    @defeature_ratio.setter
+    def defeature_ratio(self, value):
+        self._defeature_settings.DefeatureRatio = value
+        self._parent._update_setup()
+
+    @property
+    def healing_option(self):
+        """Healing option.
+
+        Returns
+        -------
+
+        """
+        return self._defeature_settings.HealingOption
+
+    @healing_option.setter
+    def healing_option(self, value):
+        self._defeature_settings.HealingOption = value
+        self._parent._update_setup()
+
+    @property
+    def model_type(self):
+        """Model type.
+
+        Returns
+        -------
+        int
+        """
+        return self._defeature_settings.ModelType
+
+    @model_type.setter
+    def model_type(self, value):
+        """Model type (General 0 or IC 1)."""
```

Comment:
Again, this maybe should be reformatted to something like:

Model type. Options are:

- 0=General
- 1=IC